src/main/java/org/motechproject/prevac/service/impl/VisitScheduleOffsetServiceImpl.java

```diff
@@ -28,38 +31,20 @@ public VisitScheduleOffset findByVisitTypeAndStageId(VisitType visitType, Long s
     }
 
     @Override
-    public Map<VisitType, VisitScheduleOffset> getAsMapByStageId(Long stageId) {
-        Map<VisitType, VisitScheduleOffset> visitScheduleOffsetMap = new HashMap<>();
-        List<VisitScheduleOffset> visitScheduleOffsetList = visitScheduleOffsetDataService.findByStageId(stageId);
-
-        if (visitScheduleOffsetList != null) {
-            for (VisitScheduleOffset offset : visitScheduleOffsetList) {
-                visitScheduleOffsetMap.put(offset.getVisitType(), offset);
-            }
-        }
-
-        return visitScheduleOffsetMap;
-    }
-
-    @Override
-    public Map<Long, Map<VisitType, VisitScheduleOffset>> getAllAsMap() {
-        Map<Long, Map<VisitType, VisitScheduleOffset>> visitScheduleOffsetMap = new HashMap<>();
-        Map<VisitType, VisitScheduleOffset> visitTypeMap;
+    public Map<VisitType, VisitScheduleOffset> getAllAsMap() {
+        Map<VisitType, VisitScheduleOffset> visitTypeMap = new HashMap<>();
         List<VisitScheduleOffset> visitScheduleOffsetList = visitScheduleOffsetDataService.retrieveAll();
 
-        if (visitScheduleOffsetList != null) {
-            for (VisitScheduleOffset offset : visitScheduleOffsetList) {
-                visitTypeMap = visitScheduleOffsetMap.get(offset.getStageId());
-
-                if (visitTypeMap == null) {
-                    visitTypeMap = new HashMap<>();
-                    visitScheduleOffsetMap.put(offset.getStageId(), visitTypeMap);
-                }
+        if (visitScheduleOffsetList == null || visitScheduleOffsetList.isEmpty()) {
+            throw new VisitScheduleException("Visit Schedule Offsets don't exist");
```

Comment:
remove this, because there should not be and exception in reschedule service (and this is used there too)